pycbc/population/rates_functions.py

```diff
@@ -0,0 +1,424 @@
+    ''' Calculate the log of background fall-off
+
+        Parameters
+        ----------
+        trigs: array
+               SNR values of all the triggers
+        bins: string
+               bins for histogrammed triggers
+        path: string
+               counts for histogrammed triggers   
+
+        Returns
+        -------
+        array
+    '''
+
+    trigs = np.atleast_1d(trigs)
+    
+    N = sum(counts)
+    
+    assert np.all(trigs >= np.min(bins)), 'cannot have triggers smaller than bin lower limit'
+    
+    # If there are any triggers that are louder than the max bin, put one 
+    # fictituous count in a bin that extends from the limits of the slide triggers
+    # out to the loudest trigger.
+    
+    # If there is no counts for a foreground trigger put a fictious count in the background bin
+    if np.any(trigs >= np.max(bins)):
+        N = N + 1
+        #log_plimit = -np.log(N) - np.log(np.max(trigs) - bins[-1])  NEEDS CHECKING
+    
+    log_rhos = []
+    for t in trigs:
+        if t >= np.max(bins):
+            log_rhos.append(-np.log(N)-np.log(np.max(trigs) - bins[-1])) 
+        else:
+            i = bisect.bisect(bins, t) - 1
+            #print t, counts[i], np.log(bins[i+1] - bins[i])
+            if counts[i] == 0:
+                counts[i] = 1
+            log_rhos.append(np.log(counts[i]) - np.log(bins[i+1] - bins[i]) - np.log(N))
+    return np.array(log_rhos)
+
+
+#def log_rho_fg_analytic(t):
+#    return np.log(3.0) + 3.0*np.log(rhomin) - 4.0*np.log(t)
+
+
+def log_rho_fg_mc(t, injstats, bins):
+    counts, bins = np.histogram(injstats, bins)
+    
+    N = sum(counts)
+    dens = counts / np.diff(bins) / float(N)
+
+    assert np.min(t) >= np.min(bins)
+    assert np.max(t) < np.max(bins)
+    
+    tinds = np.searchsorted(bins, t) - 1
+    
+    return np.log(dens[tinds])
+    
+def fit(R):
+    ''' Fit skew - lognormal to the rate samples achived from a prior analysis
+
+        Parameters
+        ----------
+        R: array
+           Rate samples
+
+        Returns
+        -------
+        ff[0]: float
+            The skewness
+        ff[1]: float
+            The mean
+        ff[2]: float
+            The standard deviation 
+    '''
+    
+    def pdf(x):
+        return 1/np.sqrt(2*np.pi) * np.exp(-x**2/2)
+
+    def cdf(x):
+        return (1 + erf(x/np.sqrt(2))) / 2
+
+    def skew(x, alpha, mu, sigma):
+        t = (x - mu) / sigma
+        return 2 / sigma * pdf(t) * cdf(alpha*t)
+
+    def optm(l, y, x):
+            alpha, mu, sigma = l
+            return kde(x)- skew(x, alpha, mu, sigma)    
+    
+    R = np.log(R)
+    xs = np.linspace(min(R), max(R), 200)
+    
+    mu_norm, sigma_norm = np.mean(R), np.std(R)
+    
+    kde = ss.gaussian_kde(R)
+    p_xs = kde(xs)
+    
+    ff, flag = optimize.leastsq(optm, [0.1, mu_norm, sigma_norm], args=(p_xs, xs))
+    return ff[0], ff[1], ff[2]
+
+# The flat in log and power-law mass distribution models  #
```

Comment:
I'd prefer if you used the distributions that are defined in the distributions module for the following instead. There, [https://github.com/ligo-cbc/pycbc/blob/master/pycbc/distributions/uniform_log.py#L22](uniform log) and [https://github.com/ligo-cbc/pycbc/blob/master/pycbc/distributions/power_law.py#L23](power law) distributions are defined. Do they do what you need? If not, can you clarify what else they would need? I'd prefer to add any addition functionality you need to those classes if possible.

Reply:
Thanks for the suggestions. I would personally prefer to keep astrophysical distributions within the population module. It might be better to add one more file models.py and populate it as we move forward in number of detections. We can talk more about this.